admin-auth/src/test/kotlin/com/epam/drill/admin/auth/UserApiKeyTests.kt

```diff
@@ -0,0 +1,117 @@
+package com.epam.drill.admin.auth
+
+import com.epam.drill.admin.auth.model.*
+import com.epam.drill.admin.auth.principal.Role
+import com.epam.drill.admin.auth.route.*
+import com.epam.drill.admin.auth.service.ApiKeyService
+import io.ktor.application.*
+import io.ktor.auth.*
+import io.ktor.features.*
+import io.ktor.http.*
+import io.ktor.locations.*
+import io.ktor.routing.*
+import io.ktor.serialization.*
+import io.ktor.server.testing.*
+import kotlinx.datetime.LocalDateTime
+import kotlinx.serialization.builtins.ListSerializer
+import kotlinx.serialization.json.Json
+import org.kodein.di.bind
+import org.kodein.di.ktor.di
+import org.kodein.di.provider
+import kotlin.test.*
+import org.mockito.Mock
+import org.mockito.MockitoAnnotations
+
+/**
+ * Testing /user-keys routers
+ */
+class UserApiKeyTests {
+    @Mock
+    lateinit var apiKeyService: ApiKeyService
+
+    @BeforeTest
+    fun setup() {
+        MockitoAnnotations.openMocks(this)
+    }
+
+    @Test
+    fun `'GET user-keys' must return the expected number of user's api keys`() {
+        val testUserId = 43
+        withTestApplication(withRoute {
+            authenticate {
+                getUserApiKeysRoute()
+            }
+        }) {
+            with(handleRequest(HttpMethod.Get, "/user-keys") {
+                addHeader(HttpHeaders.ContentType, ContentType.Application.Json.toString())
+                addJwtToken(username = "test-user", userId = testUserId)
+            }) {
+                assertEquals(HttpStatusCode.OK, response.status())
+                val response: List<ApiKeyView> = assertResponseNotNull(ListSerializer(ApiKeyView.serializer()))
+                assertEquals(2, response.size)
```

Comment:
I don't see the part where we populate repository width data or mock it. Am I missing something?

Reply:
not implemented yet